src/pages/academy/academyRoutes.tsx

```diff
@@ -0,0 +1,108 @@
+
+const getCommonAcademyRoutes = (
+  assessmentConfigurations: AssessmentConfiguration[]
+): RouteObject[] => {
+  const assessmentRoutes =
+    assessmentConfigurations?.reduce((acc: Record<string, AssessmentConfiguration>, config) => {
+      acc[assessmentTypeLink(config.type)] = config;
+      return acc;
+    }, {}) ?? {};
+
+  const assessmentLoader: LoaderFunction = ({ params }) => {
+    const requestedType = params['assessmentConfigType'];
+    for (const type of Object.keys(assessmentRoutes)) {
+      if (requestedType == type) {
+        return assessmentRoutes[type];
+      }
+    }
+    return redirect(notFoundPath);
+  };
+
+  const homePageRedirect = () => {
+    const { role, enableGame } = store.getState().session;
+    if (enableGame) {
+      return redirect('game');
+    }
+    if (assessmentConfigurations && assessmentConfigurations.length > 0) {
+      return redirect(`${assessmentTypeLink(assessmentConfigurations[0].type)}`);
+    }
+    if (role === Role.Admin) {
+      return redirect('adminpanel');
+    }
+    return null;
+  };
+
+  const gameRoute = new GuardedRoute({ path: 'game', lazy: Game })
+    .check(s => !!s.session.enableGame, notFoundPath)
+    .build();
+
+  return [
+    gameRoute,
+    { path: '', element: <Navigate replace to={notFoundPath} />, loader: homePageRedirect },
+    {
+      path: `:assessmentConfigType/${assessmentRegExp}`,
+      element: <Assessment />,
+      loader: assessmentLoader
+    },
+    { path: 'notipreference', lazy: NotiPreference },
+    { path: 'sourcecast/:sourcecastId?', lazy: Sourcecast },
+    { path: 'achievements/*', lazy: Achievement },
+    { path: notFoundPath, lazy: NotFound },
+    { path: '*', loader: () => redirect(notFoundPath) }
+  ];
+};
+
+const GroundControl = () => import('./groundControl/GroundControlContainer');
+const Grading = () => import('./grading/Grading');
+const Sourcereel = () => import('./sourcereel/Sourcereel');
+const GameSimulator = () => import('./gameSimulator/GameSimulator');
+const TeamFormation = () => import('./teamFormation/TeamFormation');
+const TeamFormationForm = () => import('./teamFormation/subcomponents/TeamFormationForm');
+const TeamFormationImport = () => import('./teamFormation/subcomponents/TeamFormationImport');
+const Dashboard = () => import('./dashboard/Dashboard');
+
+const staffRoutes: RouteObject[] = [
+  { path: 'groundcontrol', lazy: GroundControl },
+  { path: `grading/${gradingRegExp}`, lazy: Grading },
+  { path: 'sourcereel', lazy: Sourcereel },
+  { path: 'gamesimulator', lazy: GameSimulator },
+  { path: 'teamformation', lazy: TeamFormation },
+  { path: 'teamformation/create', lazy: TeamFormationForm },
+  { path: `teamformation/edit/${teamRegExp}`, lazy: TeamFormationForm },
+  { path: 'teamformation/import', lazy: TeamFormationImport },
+  { path: 'dashboard', lazy: Dashboard }
+].map(r => new GuardedRoute(r).check(s => s.session.role !== Role.Student, notFoundPath).build());
```

Comment:
Let's change this to a whitelist instead.

Reply:
Resolved in 90dba2e1f75fad50fc5842915413181fa44dc5b0.